DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/DllConflictManagment/DllConflictManager.cs

```diff
@@ -0,0 +1,245 @@
+    {
+      if (assembly.GetType(typeName) is Type foundType)
+      {
+        type = foundType;
+        return true;
+      }
+    }
+    return false;
+  }
+
+  public bool HandleTypeLoadException(TypeLoadException ex)
+  {
+    // getting private fields is a bit naughty..
+    var assemblyName = GetPrivateFieldValue<string?>(ex, "AssemblyName")?.Split(',').FirstOrDefault();
+
+    StringBuilder sb = new();
+    if (assemblyName != null && _assemblyConflicts.TryGetValue(assemblyName, out var assemblyConflictInfo))
+    {
+      sb.AppendLine($"Speckle encountered a dependency mismatch error.");
+      sb.AppendLine();
+      sb.AppendLine($"Dependency Name: {assemblyName}");
+      sb.AppendLine($"Expected Version: {assemblyConflictInfo.SpeckleDependencyAssemblyName.Version}");
+      sb.AppendLine($"Actual Version: {assemblyConflictInfo.ConflictingAssembly.GetName().Version}");
+      sb.AppendLine();
+      sb.AppendLine($"Conflicting dll folder: {assemblyConflictInfo.GetConflictingExternalAppName()}");
+
+      TaskDialog.Show("Conflict Report 🔥", sb.ToString());
+      return true;
+    }
+
+    return false;
+  }
+
+  private static T GetPrivateFieldValue<T>(object obj, string fieldName)
+  {
+    if (obj == null)
+    {
+      throw new ArgumentNullException("obj");
+    }
+
+    FieldInfo fi =
+      obj.GetType().GetField(fieldName, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+      ?? throw new ArgumentOutOfRangeException(
+        "propName",
+        string.Format("Property {0} was not found in Type {1}", fieldName, obj.GetType().FullName)
+      );
+
+    return (T)fi.GetValue(obj);
+  }
+
+  public void WarnUserOfPossibleConflicts()
+  {
+    if (_assemblyConflicts.Count == 0)
+    {
+      return;
+    }
+
+    var options = _optionsLoader.LoadOptions();
+    List<AssemblyConflictInfo> conflictInfoNotIgnored = _assemblyConflicts.Values
+      .Where(info => !options.DllsToIgnore.Contains(info.SpeckleDependencyAssemblyName.Name))
+      .ToList();
+
+    if (conflictInfoNotIgnored.Count == 0)
+    {
+      return;
+    }
+
+    StringBuilder sb = new();
+    foreach (var assemblyConflictInfo in conflictInfoNotIgnored)
+    {
+      sb.AppendLine($"Dependency Name:  {assemblyConflictInfo.SpeckleDependencyAssemblyName.Name}");
+      sb.AppendLine($"Expected Version: {assemblyConflictInfo.SpeckleDependencyAssemblyName.Version}");
+      sb.AppendLine($"Actual Version:   {assemblyConflictInfo.ConflictingAssembly.GetName().Version}");
+      sb.AppendLine($"Conflicting version folder: {assemblyConflictInfo.GetConflictingExternalAppName()}");
+      sb.AppendLine();
+    }
+
+    using var dialog = new TaskDialog("Conflict Report 🔥");
+    dialog.MainInstruction =
+      "Speckle encountered an error loading the following dependencies. This is likely due to a different addin that uses a different version of the same dependency.";
```

Comment:
At this point, is this definitely an error?

Reply:
Good point. It was an error loading the dependency that we needed, but it may not actually be an issue. I'll change the wording